### ISSUE TITLE
Troubles with batch inserting

### ISSUE THREAD

**Issue by HardWin13** [AdysTech/InfluxDB.Client.Net#59]
@mvadu, thanks for nice library. 

I have several questions about your batch inserting algoritm.
So, I develop application that collects and posts big data in InfluxDb regularly. And I need to insert about 100 000 points as soon as possible. 
This operation takes about 5 seconds in my computer when I use your library method `PostPointsAsync (dbName, points) ` and it's so long for me.

I tried to write my own test and insert 100 000 points in InfluxDb directly due to line protocol:

```cs
var queue = new Queue<Point>();
var data = new StringBuilder();
for (var i = 1; i <= 100000; i++)
{
	var point = new Point("test", DateTime.UtcNow.AddSeconds(i), 50, 50, 50);
	queue.Enqueue(point);
	data.Append($"{point.LoopName} SP={point.SP},PV={point.PV},OP={point.OP} {1434055562000000000+i}\n ");
}
				
var dataString = data.ToString();

using (var client = new HttpClient())
{
	client.BaseAddress = new Uri("http://localhost:8086/write?db=smkr_db");
	var content = new ByteArrayContent(Encoding.ASCII.GetBytes(dataString));
	var result = client.PostAsync(client.BaseAddress,content);
	result.Wait();
}
```
This operation took about 0.6 seconds and it's great)

Thereafter, I observed that you post data batch in size of 255 points. I think it's main reason that big data inserting takes so much time.

So, my questions:
1) Why do you post batch per 255 points? Is that some kind of magic number for InfluxDb?
2) What are the advantages of your approach and disadvantages of my test code above?
3) Perhaps it will be better to change your method like this:

```cs
public async Task<bool> PostPointsAsync(string dbName, IEnumerable<IInfluxDatapoint> Points, int maxBatchSize = 255)
```
I think it will be useful for client to be able to change the batch size in his way.

Thanks in advance!

**Comment by mvadu** [comment]
@HardWin13 thank you for this topic. It was one of the old traits.. I did lot of work in making sure InfluxDB (actual DB side, go lang compatibility with windows way of handling files.) works with Windows back in the days. That time due to stability issues I restricted batch size to 255 points. Whole idea of having that as an [variable](https://github.com/AdysTech/InfluxDB.Client.Net/blob/23a2d9810db22263477e60229347ef4346b2074e/src/DataStructures/InfluxDBClient.cs#L414) was to expose it to consumer. Never got to do it. Thank you for bringing it up. I will fix it in few days.
